specs/agents-versioning-publishing.md

```diff
@@ -0,0 +1,62 @@
+# Hypertrace Agents versioning and publishing
+
+This document describes versioning and publishing requirements for Hypertrace agents.
+All Hypertrace agents follow these requirements:
+
+* Semantic versioning for all released artifacts.
+* Release every merge commit into the main or maintenance branch as `"snapshot"`.
+* Maintenance branches for older versions should be named as `release-<major>.<minor>` e.g. `release-1.0`.
+
+#### Semantic versioning
+
+Each released version strictly follows [semantic versioning](https://semver.org/).
+The released version must be tagged in the source code: e.g. `1.0.0` or `v1.0.0` for languages
+with this convention (e.g. Golang).
+
+#### Snapshot release for every merge commit
+
+Each commit merged into the `main` or maintenance branch (e.g. `release-1.1` for `1.1.x`) should be
```

Comment:
this is very strange. this implies a README will eat build/dev/deploy/rate-limit resources

In Zipkin, we've had numerous rate limits hit, including total size of archives, downloads and cardinality of artifacts. This is surprising to make such fine grained promises.. Why not simply discuss releases as in versioned releases and everything else undefined /at own risk?

Reply:
As far as I can tell this is Traceable's requirement. We can make this as a recommendation and then agents maintainers will ultimately decide how to do this.

Reply:
This is mainly a constraint in java world and maybe npm (but that is less problematic) as other languages whose versioning works on top of for example git have access to this anyways (e.g. Go or PHP).